util/pkg/vfs/s3context.go

```diff
@@ -146,10 +148,10 @@ func (s *S3Context) getDetailsForBucket(bucket string) (*S3BucketDetails, error)
 	}
 
 	// Probe to find correct region for bucket
-	endpoint := os.Getenv("S3_ENDPOINT")
+	endpoint := viper.GetString("S3_ENDPOINT")
```

Comment:
Calling `viper.BindEnv()` should make env reading possible. A little annoying, but not sure how to go about it differently at the moment.
```suggestion
	_ = viper.BindEnv("S3_ENDPOINT")
	endpoint := viper.GetString("S3_ENDPOINT")
```